qiskit/providers/ibmq/job/ibmq_composite_job.py

```diff
@@ -0,0 +1,1274 @@
+                                            removal_tags=removal_tags)
+        for job in self._get_circuit_jobs():
+            tags_to_update = new_tags.union(
+                {tag for tag in job.tags() if tag.startswith(self._tag_prefix)})
+            auto_retry(job.update_tags, list(tags_to_update))
+        self._tags = list(new_tags)
+        return self._tags
+
+    def _get_tags_to_update(self,
+                            replacement_tags: Optional[List[str]],
+                            additional_tags: Optional[List[str]],
+                            removal_tags: Optional[List[str]]) -> Set[str]:
+        """Create the list of tags to update for this job.
+
+        Args:
+            replacement_tags: The tags that should replace the current tags
+                associated with this job.
+            additional_tags: The new tags that should be added to the current tags
+                associated with this job.
+            removal_tags: The tags that should be removed from the current tags
+                associated with this job.
+
+        Returns:
+            The new tags to associate with this job.
+
+        Raises:
+            IBMQJobInvalidStateError: If any of the input parameters are invalid.
+        """
+        tags_to_update = set(self._tags or [])  # Get the current job tags.
+        if isinstance(replacement_tags, list):  # `replacement_tags` could be an empty list.
+            # Replace the current tags and re-add those associated with a job set.
+            validate_job_tags(replacement_tags, IBMQJobInvalidStateError)
+            tags_to_update = set(replacement_tags)
+        if additional_tags:
+            # Add the specified tags to the tags to update.
+            validate_job_tags(additional_tags, IBMQJobInvalidStateError)
+            tags_to_update.update(additional_tags)
+        if removal_tags:
+            # Remove the specified tags, except those related to a job set,
+            # from the tags to update.
+            validate_job_tags(removal_tags, IBMQJobInvalidStateError)
+            for tag_to_remove in removal_tags:
+                if tag_to_remove in tags_to_update:
+                    tags_to_update.remove(tag_to_remove)
+                else:
+                    logger.warning('The tag "%s" for job %s will not be removed, because it was '
+                                   'not found in the job tags to update %s',
+                                   tag_to_remove, self.job_id(), tags_to_update)
+        return tags_to_update
+
+    def status(self) -> JobStatus:
+        """Query the server for the latest job status.
+
+        Note:
+            This method is not designed to be invoked repeatedly in a loop for
+            an extended period of time. Doing so may cause the server to reject
+            your request.
+            Use :meth:`wait_for_final_state()` if you want to wait for the job to finish.
```

Comment:
`wait_for_final_state()` already takes a callback for real time status update. We cannot change the behavior of `status()` because it's dictated by the parent class `Job`.